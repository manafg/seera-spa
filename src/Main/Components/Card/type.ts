type CardProps = {
  /**
   * describe the name of the hotel in the card
   */
  name: string;
  /**
   * describe the price for the hotel in the card
   */
  price: string;
  /**
   * describe the city that hotel in
   */
  city: string;
};

export type { CardProps };
