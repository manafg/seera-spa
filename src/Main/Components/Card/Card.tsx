import { StarIcon } from "@chakra-ui/icons";
import { Badge, Image, Box } from "@chakra-ui/react";
import React from "react";

import { Deal } from "../../Context/DealsContext/type";

function Card({ name, price, city, available_on }: Deal) {
  const property = {
    imageUrl: "https://bit.ly/2Z4KKcF",
    imageAlt: "Rear view of modern home with pool",
    beds: 3,
    baths: 2,
    title: "Modern home in city center in the heart of historic Los Angeles",
    formattedPrice: "$1,900.00",
    reviewCount: 34,
    rating: 4,
  };

  return (
    <Box
      maxW='sm'
      borderWidth='1px'
      borderRadius='lg'
      bg='white'
      overflow='hidden'
    >
      <Image src={property.imageUrl} alt={property.imageAlt} />

      <Box p="6">
        <Box d="flex" alignItems="baseline">
          <Badge borderRadius="full" px="2" colorScheme="red">
            New
          </Badge>
          <Box
            color='gray.500'
            fontWeight="semibold"
            letterSpacing="wide"
            fontSize='xs'
            textTransform="uppercase"
            ml='2'
          >
            Available on : {available_on}
          </Box>
        </Box>

        <Box
          mt="1"
          fontWeight="semibold"
          as='h4'
          lineHeight="tight"
          isTruncated={true}
        >
          {name}
        </Box>

        <Box>
          ${price}
          <Box as="span" color="gray.600" fontSize="sm">
            / wk
          </Box>
        </Box>

        <Box d="flex" mt="2" alignItems="center">
          {Array(5)
            .fill("")
            .map((_, i) => (
              <StarIcon
                key={i}
                color={i < property.rating ? "red.400" : "gray.300"}
              />
            ))}
          <Box as="span" ml="2" color="gray.600" fontSize="sm">
            City:   {city}
          </Box>

        </Box>
      </Box>
    </Box>
  );
}

export default Card;
