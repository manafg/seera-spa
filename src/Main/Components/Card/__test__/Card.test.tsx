import { render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";

import Card from "../Card";

import "@testing-library/jest-dom/extend-expect";
import rrender from "react-test-renderer";

describe("Snapshot tests", () => {
  test("Matches existing Snapshot", () => {
    const tree = rrender
      .create(
        <Card
          name='taName'
          price='dasdsa'
          available_on='25/10/2019'
          city='amman'
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
