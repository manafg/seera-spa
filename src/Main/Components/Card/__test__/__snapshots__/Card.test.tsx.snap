// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Matches existing Snapshot 1`] = `
<div
  className="css-gklzgs"
>
  <img
    alt="Rear view of modern home with pool"
    className="chakra-image css-0"
    src="https://bit.ly/2Z4KKcF"
  />
  <div
    className="css-1vae7t2"
  >
    <div
      className="css-x1sij0"
    >
      <span
        className="chakra-badge css-9doude"
      >
        New
      </span>
      <div
        className="css-qvxrcx"
      >
        Available on : 
        25/10/2019
      </div>
    </div>
    <h4
      className="css-1u4mx17"
    >
      taName
    </h4>
    <div
      className="css-0"
    >
      $
      dasdsa
      <span
        className="css-bgvdb6"
      >
        / wk
      </span>
    </div>
    <div
      className="css-15yujhg"
    >
      <svg
        className="chakra-icon css-4i8mdq"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
      <svg
        className="chakra-icon css-4i8mdq"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
      <svg
        className="chakra-icon css-4i8mdq"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
      <svg
        className="chakra-icon css-4i8mdq"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
      <svg
        className="chakra-icon css-1fdnb51"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="css-ekuaes"
      >
        City:   
        amman
      </span>
    </div>
  </div>
</div>
`;
